### PR TITLE
feat: changed permission escalation of database owner to iam_creator because of postgres 15 change

### PR DESCRIPTION
**Pull request by kjuulh**
This is because of this line in the postgresql 15 release notes:

> Remove the default ADMIN OPTION privilege a login role has on its own role membership (Robert Haas)
> 
> Previously, a login role could add/remove members of its own role, even without ADMIN OPTION privilege.

https://arc.net/l/quote/ddxhijwr

Backup link: https://www.postgresql.org/docs/release/15.0/

--- 

The gist of the change is that previously we relied on the fact that a login user implicitly had admin option on their own role.

This is no longer possible in postgresql 15, because they fixed a bug, where normally you cannot assign admin option to itself, because a role cannot be a member of itself. It was fixed because assigning a role onto others without superuser, is technically privilige escalation.

A side effect is that ALTER and DROP which are only allowed to be performed by a database owner, can no longer be transferred by said owner itself, with it being a superuser.

As such we instead change the code to use the superuser (iam_creator) to perform this change.

I.e.

```sql

-- login: <name>
GRANT <name> TO <name>_readowningwrite

-- to

-- login: <iam_creator>
GRANT <name> to <name>_readowningwrite
```

This change should only affect those setup with `extended-writes`, which should only be user for PostgresUsers in environments where destructive changes are negligible, such as dev environments

### CODE REVIEW

**Comment by kjuulh** [comment]
The pr is very much in progress. Not all the tests pass after I'd changed passwords from "" to something else, (required in later versions of postgres). 

Also there is a few more `WITH ADMIN OPTION` than is needed, I haven't tested it yet, but only the sqlConnection -> db should be required.

File: docker-compose.yml [hunk 1]
```diff
@@ -1,9 +1,10 @@
 version: '2'
 services:
   postgres:
-    image: timms/postgres-logging:11.5
+    image: timms/postgres-logging:15.0
```
Comment (mahlunar) [inline docker-compose.yml:4]: Let's also check 16.1. I think you can just use `postgres:16.1-alpine`

Comment (kjuulh) [reply docker-compose.yml:4]: Times didnt have pne available i think but i will snap it for something else

**Comment by kjuulh** [comment]
@mahlunar I had to remove a test, 15 changed how public works, so the test no longer made sense

**Review by mahlunar — APPROVED** [review]

**Pull request merged**